### PR TITLE
MOTECH-2020 Refactored task field bubbles

### PR DESCRIPTION
**Pull request by nickdotreid**
This is an initial commit to get the code review process going. There are minor bugs when interacting with the field bubble popover, and large chunks of now legacy code that will be removed.

### CODE REVIEW

**Comment by nickdotreid** [comment]
test this please

**Comment by sebbrudzinski** [comment]
@nickdotreid The description in the pull requests says that there are some bugs when interacting with the field bubble popover. Is this still valid or have the problems been resolved yet? If not, do we have a ticket that handles the fixes?

**Comment by nickdotreid** [comment]
Yes there are still bugs // I'm going to change the popover interaction to
a modal window (which will fix it) -- don't merge yet, there is one more
commit coming

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 1]
```diff
@@ -537,719 +508,255 @@
...
-                    var manipulateElement = $("[ismanipulate=true]"),
-                        manipulation = "dateTime(" + $("#dateFormat").val() + ")",
-                        elementManipulation = manipulateElement.attr("manipulate"),
-                        regex = new RegExp("dateTime\\(.*?\\)", "g");
-
-                    elementManipulation = elementManipulation.replace(regex, manipulation);
-                    manipulateElement.attr("manipulate", elementManipulation);
-                    manipulateElement.trigger('manipulateChanged');
-                });
+                if(attrs.active) {
```
Comment (sebbrudzinski) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: This if condition is exactly the same as the previous condition just few lines above. Can those ifs be merged into one?

**Comment by sebbrudzinski** [comment]
Thanks @nickdotreid 
Can you please let me know when this is finished and ready to merge? (removing the "Do not merge" label is sufficient)

**Comment by nickdotreid** [comment]
test this please

**Comment by nickdotreid** [comment]
@sebbrudzinski @pgesek // this should be ready for merge and review once the test are complete

**Comment by sebbrudzinski** [comment]
@nickdotreid There are conflicts on this branch that must be resolved

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js (hunk 1 above)
Comment (nickdotreid) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: PS, @sebbrudzinski  just saw this and combined the if statements...

**Comment by sebbrudzinski** [comment]
@aczyrko, we need to test that after this change, the Tasks module still works properly. This pull request does not bring new features, but it contains a huge refactor of the module, so please verify that creating and editing tasks still works properly. We should focus on testing the draggable bubbles and manipulations.

**Comment by aczyrko** [comment]
@nickdotreid During preparation of testing PR conflicts occurred, please solve them and merge to the highest version.

**Comment by nickdotreid** [comment]
@aczyrko -- this PR is up to date, and while I've done some testing.... it probably doesn't hurt to test it again before merging

**Comment by nickdotreid** [comment]
@aczyrko actually I take that back... I thought I could get away with not refactoring the Boolean field bubble... and now that I tested that, things be breaking...

**Comment by nickdotreid** [comment]
@sebbrudzinski @aczyrko 

I updated the PR to catch boolean fields and treat them correctly, and I made 2 other smaller changes. I have not squashed these commits so they are easy to see/review. I'm happy to squash them before merging (if that makes sense).

There are issues with the placeholder text that seem to be from before my refactor (meaning you can focus-in then focus-out, and the field will treat the placeholder text as a value). I'd like to get this fixed up before merging....

**Comment by nickdotreid** [comment]
@aczyrko @sebbrudzinski 

This PR should be ready for testing and merging... I fixed the placeholder text issue I created, I think there is another issue there that also needs resolving in a separate ticket...

**Comment by aczyrko** [comment]
**Errors**
- I can enter only one "bean".
- Possible errors setting has higher priority than Retry Count in Retry task on failure field.
- Incorrect display of retried tasks in task activity.
- When I enter this same „bean” two times „bean” is not available only {{bean}} format.


More information available in attachment.
[PR357.pdf](https://github.com/motech/motech/files/212378/PR357.pdf)

**Comment by nickdotreid** [comment]
@aczyrko I got your bugs... I'll get to them in the next day or so and will work to fix them...

I'll take a look at MOTECH-2318 (I'm pretty sure I never touched that file, but I'll look through the commit and see if it didn't get merged somehow...

**Comment by pgesek** [comment]
@nickdotreid  I would say that the retry bugs are out of scope for this, since they are covered by different tickets

**Comment by nickdotreid** [comment]
@aczyrko @sebbrudzinski 

I fixed problems A & C, the change should be visible in the last commit (and should be pretty funny to look at the delta)

**Comment by aczyrko** [comment]
@nickdotreid thank you for information! I confirm that problems A (I can enter only one „bean”) and C  (When I enter this same „bean” two times „bean” is not available only {{bean}} format) are fixed. 

More details in attachment.
[PR357(1).pdf](https://github.com/motech/motech/files/216604/PR357.1.pdf)

**Comment by pgesek** [comment]
@nickdotreid I will merge this after you resolve the conflicts

**Comment by nickdotreid** [comment]
@pgesek its mergeable!

**Pull request merged**